core/server/api/index.js

```diff
@@ -209,6 +209,10 @@ http = function http(apiMethod) {
             // Add X-Cache-Invalidate, Location, and Content-Disposition headers
             return addHeaders(apiMethod, req, res, (response || {}));
         }).then(function then(response) {
+            if (req.method === 'DELETE') {
+                return res.status(204).end();
```

Comment:
This could use [sendStatus](http://expressjs.com/en/4x/api.html#res.sendStatus), I think?